master/internal/command/command.go

```diff
@@ -235,9 +237,15 @@ func (c *Command) OnExit(ae *task.AllocationExited) {
 
 	c.exitStatus = ae
 
-	if err := c.db.CompleteTask(c.taskID, time.Now().UTC()); err != nil {
+	err := c.db.CompleteTask(c.taskID, time.Now().UTC())
+	if err != nil {
+		var sErr status.Status
+		if errors.As(err, &sErr) && sErr.Code() != codes.NotFound {
+			return
```

Comment:
I totally blanked reading this check; I thought we discussed putting it somewhere else, I don't think CompleteTask and even return a status.Status